### DISCUSSION TITLE
Make SKP Compatible with Pico W

### DISCUSSION THREAD

**Discussion by mwedmark** [frntc/SIDKick-pico#9] [General]
I built the firmware with small changes for a Pico W and tried it in one of my machines. No can do. These is no sound and SYS do not work for starting the settings app. This is my first built SKP so I could have other problem... 

Not sure why it is not working though..
I don't use DAC's yet and LED is on another pin so I made these changes:
Commented out
//#define USE_DAC

and commented out
//#define FLASH_LED

As new code for flashing would need to be specifically written for Pico W where the LED (for some reason probably saving of pins) is controlled by WIFI chip? "Blink" app for Pico W was used as a reference for tool-chain and works.

Also had some kind of error mentioned in other thread with:
expression must have a constant value

So I changed some values around 721-727 from being char array pointers to specific bytes in array to the same constant value.
It now looks like this:

-snip-
uint16_t launcherAddress = (0x03 << 8 ) + 0x3C;
uint8_t *transferData	 = (uint8_t *)&launchCode[2],
		*transferDataEnd = (uint8_t *)&launchCode[launchSize];
uint16_t jumpAddress     = 0xD401;
uint8_t  transferReg[ 32 ] = {
	0x78, 0x48, 0x68, 0xA9, 0xA9, 0x48, 0x68, 0x8D, 
	 0x3C, 0x03, 0x48, 0x68, 0x4C, 0x01, 0xD4 };
-snip-

Any ideas? is there a mode where one could control the SID via UART via USB? That would be cool, like a debug mode on the bench. There might even be programs on PC that supports SID's like that..

**Comment by magwed** [comment]
I accidently soldered my PicoW backwards... :-) I will try and resolder it and try again. When using the Pi Pico (without W) is much easier to get it right as 3 debug points at the short side should add up for both Pi Pico and the Sidkick PCB. I also tested it in a C64 PCB not like the pictures (older NTSC) which made it even harder to follow the pictures.
